python/tchannel/tornado/peer.py

```diff
@@ -500,7 +500,7 @@ def send(self, arg1, arg2, arg3,
         arg1, arg2, arg3 = (
             maybe_stream(arg1), maybe_stream(arg2), maybe_stream(arg3)
         )
-
+        assert attempt_times != 0
```

Comment:
No. Not here. This is not what I was saying. This will break things. You call `_send_with_retry` from this function. Add the assert in `_send_with_retry`.

The point of adding the statement there is that if it ever gets called with `attempt_times < 1`, we know that we made a mistake in the code that calls it. Otherwise it'll be very difficult to debug why a request isn't going through.